common/core/gpudevice.cpp

```diff
@@ -652,8 +659,6 @@ void GpuDevice::InitializeHotPlugEvents(bool take_lock) {
 }
 
 void GpuDevice::HandleRoutine() {
-  thread_sync_lock_.lock();
```

Comment:
Any reason to change this logic can we keep it the same ? This allows HandlwHWCSettings to be executed from worker or main thread, which ever gets their first.

Reply:
If I remember correctly from debug several weeks ago, calling HandleHWCSettings() again results in HWC crash (Not this patch, an intermediate patch). Disable this function call is to avoid HWC crash. And yes, I believe if we have an active thread and could load user setting timely without reboot, it could enable much flexibility to configure HWC instantly. But to make this works, it requires some debug & tuning to make it works as expected and how to design thread loop is important too. So far the thread bundled with GpuDevice is one time use and will fall into sleep after /vendor/hwc.lock exclusive lock grabbed.

Reply:
shouldn't be. If you not in HandleHWCSettings, takes a lock at start and set's it's state. If state is set it doesn't try to do it again. So, either case it should happen only once.